### PR TITLE
Pwsh create script support + pwsh-azure image

### PR DESCRIPTION
**Pull request by glicht**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/23237

## Description
update create docker image script to support powershell. Create an pwsh-azure image as an example. See README for details.

### CODE REVIEW

File: docker/build_docker.sh [hunk 1]
```diff
@@ -242,6 +248,7 @@ if [[ ! $(which pyenv) ]] && [[ -n "${CIRCLECI}" ]]; then
     export PATH="$HOME/.pyenv/bin:$PATH"
     eval "$(pyenv init -)"
     eval "$(pyenv virtualenv-init -)"
+    pyenv shell system $(pyenv versions --bare | grep 3.7)
```
Comment (Itay4) [inline docker/build_docker.sh:251]: why 3.7 and not 3.8?

File: README.md [hunk 2]
```diff
@@ -31,31 +31,36 @@ The script `docker/build_docker.sh` is used to build all modified docker images.
 * Install pipenv globally using: `pip install pipenv`
 * Install requests globally: `pip install requests`
 
-To get up and running fast with a python image with additional packages use the script: `docker/create_new_python_image.py`. Usage:
+To get up and running fast with a python image with additional packages use the script: `docker/create_new_docker_image.py`. Usage:
```
Comment (Itay4) [inline README.md:34]:
```suggestion
To get up and running fast with a Python/PowerShell image with additional packages use the script: `docker/create_new_docker_image.py`. Usage:
```

File: docker/verify_licenses.py [hunk 3]
```diff
@@ -84,6 +140,27 @@ def main():
...
+    parser.add_argument(
+        "docker_image", help="The docker image with tag version to use. For example: demisto/python3:1.5.0.27")
+    args = parser.parse_args()
+    with open('{}/approved_licenses.json'.format(sys.path[0])) as f:
+        licenses = json.load(f)["licenses"]
+    with open("{}/packages_license_check_exclude.json".format(sys.path[0])) as f:
+        ignore_packages = json.load(f)["packages"]
+    with open("{}/known_licenses.json".format(sys.path[0])) as f:
+        known_licenses = json.load(f)["packages"]
+    print("================= Checking python packages =================")
```
Comment (Itay4) [inline docker/verify_licenses.py:158]:
```suggestion
    print("================= Checking Python packages =================")
```

File: docker/verify_licenses.py [hunk 4]
```diff
@@ -84,6 +140,27 @@ def main():
...
+    check_python_license(args.docker_image, licenses,
+                         ignore_packages, known_licenses)
+    print("================= Checking Powershell packages =================")
```
Comment (Itay4) [inline docker/verify_licenses.py:161]:
```suggestion
    print("================= Checking PowerShell packages =================")
```

File: docker/create_new_docker_image.py [hunk 5]
```diff
@@ -92,33 +137,17 @@ def main():
...
-        cmd_arr = ["pipenv", "install"]
-        cmd_arr.extend(args.pkg)
-        subprocess.call(cmd_arr, cwd=folder, env=my_env)
+    if args.type == "python":
+        create_python_image(folder, base_image_last, args)
     else:
-        subprocess.call(["pipenv", "lock"], cwd=folder, env=my_env)
+        create_powershell_image(folder, base_image_last, args)
     print('Adding: {} to .dependabot/config.yml ...'.format(folder))
     subprocess.check_call([sys.path[0] + "/add_dependabot.sh", "docker/" + args.name])
```
Comment (Itay4) [inline docker/create_new_docker_image.py:148]: we can pass here the type as well so it wont print some thing `Not adding python dependency config as docker/<IMAGE NAME>/Pipfile not found` for powershell images

**Comment by Itay4** [comment]
created dockerimage as follows: 
`./docker/create_new_docker_image.py -t powershell --pkg PoshWSUS pwsh-poshwsus`


Trying to build it fails:
```
 ./docker/build_docker.sh pwsh-poshwsus
CIRCLE_SHA1 not set. Assuming local testing.
default python versions: 
Python 3.8.1
Python 3.8.1
pyenv versions:
  system
  3.7.5
  3.8.0
* 3.8.1 (set by /Users/ikeren/.pyenv/version)
DOCKER_ORG: devtesting, DIFF_COMPARE: [ALL], SCRIPT_DIR: [./docker], CIRCLE_BRANCH: , PWD: [/Users/ikeren/dev/demisto/dockerfiles]
=============== Thu Apr 16 10:23:24 IDT 2020: Starting docker build in dir: ./docker/pwsh-poshwsus ===============
Starting build for dir: ./docker/pwsh-poshwsus, image: pwsh-poshwsus, pwd: /Users/ikeren/dev/demisto/dockerfiles
pwsh-poshwsus: using version: 1.0.0.1587021804
Sending build context to Docker daemon  3.072kB
Step 1/6 : FROM demisto/powershell:6.2.4.6166
 ---> 1cf9e910b9e3
Step 2/6 : RUN pwsh -c "Set-PSRepository -Name PSGallery -InstallationPolicy Trusted -ErrorAction Stop"
 ---> Running in a6a35718057e
Set-PSRepository : No repository with the name 'PSGallery' was found.
At line:1 char:1
+ Set-PSRepository -Name PSGallery -InstallationPolicy Trusted -ErrorAc ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (PSGallery:String) [Set-PSRepository], InvalidOperationException
+ FullyQualifiedErrorId : RepositoryNotFound,Set-PSRepository
 
The command '/bin/sh -c pwsh -c "Set-PSRepository -Name PSGallery -InstallationPolicy Trusted -ErrorAction Stop"' returned a non-zero code: 1
```

am i doing something wrong?

File: docker/build_docker.sh (hunk 1 above)
Comment (glicht) [reply docker/build_docker.sh:251]: I want to make sure the script is compatible with 3.7

File: docker/create_new_docker_image.py (hunk 5 above)
Comment (glicht) [reply docker/create_new_docker_image.py:148]: Let's leave this to a separate PR.

**Pull request merged**